### DISCUSSION TITLE
Something in latest iPXE prevents Mellanox Connect X4 to set link up and boot further

### DISCUSSION THREAD

**Discussion by astekolshchikov** [ipxe/ipxe#1100] [General]
I build iPXE from latest master and hit a (possibly) bug, that prevents Mellanox Connect X4 to set link up and boot further.
Supermicro server (latest BIOS/IPMI), Mellanox Connect X4 (latest fw 14.32.1010). Tried both legacy and UEFI boot modes.
Server starts, get IP from DHCP, boot undionly.kpxe (ipxe.efi), 
then system stuck with "Error: No configuration methods succeeded (Error code 040ee1)"
As described here https://ipxe.org/err/040ee1 i build PXE with the debug option DEBUG=ifmgmt , boot new build, being dropped to iPXE shell
and from iPXE shell i cannot set interfaces up. From the switch side links are down and no MACs on ports.
Temporarily problem was solved by reverting to more than year old sources (Sept 2022), iPXE built  from old sources works well 
and not showing any strange behavior like latest sources build.
Can anyone confirm that problem persist and point me to right direction to solve it?

**Comment by NiKiZe** [comment]
You might be seeing the same as https://github.com/ipxe/ipxe/issues/1091
Even if you are not, please follow the steps for a git bisect and see if we can narrow down what the issue might be.

**Comment by astekolshchikov** [comment]
Looks like somethig is weird with latest sources as i found two issues that looks same as my
https://github.com/ipxe/ipxe/issues/1099
https://github.com/ipxe/ipxe/issues/1097

**Reply by NiKiZe** [reply]
Both of these are about NICs that don't have native drivers in iPXE, and is not related to any kind of regression.

**Comment by astekolshchikov** [comment]
@NiKiZe  I also can bisect issue to commit 8b14652e
Built from 56cc61a1 works fine, 
56cc61a1 - Michael Brown, 3 months ago : [eap] Define a supplicant model for EAP and EAPoL
built from 8b14652e - shows same behavior as built from master
8b14652e - Michael Brown, 3 months ago : [eapol] Send EAPoL-Start packets to trigger EAP authentication

**Comment by astekolshchikov** [comment]
Hope someone points me to solution, as problem still persists in 2025.
One more trouble - on newer systems such as Ubuntu 22.04 - 24.04 ipxe reverted to commit https://github.com/ipxe/ipxe/commit/56cc61a168820c7cbbe23418388129ec11699a8c cannot be built w/o manual intervention due to Error: 64bit mode not supported on 'i386' error while building make bin-x86_64-efi/ipxe.efi
I found this https://github.com/ipxe/ipxe/commit/6ca597eee.patch, hope it works...

**Comment by mcb30** [comment]
@astekolshchikov  These issues for Mellanox cards were fixed in commit https://github.com/ipxe/ipxe/commit/c11734eee (if using the native driver) and in commit https://github.com/ipxe/ipxe/commit/926816c58 (if using `snponly.efi`).
